libs/utils/trace.py

```diff
@@ -373,89 +361,64 @@ def _scanTasks(self, df, name_key='comm', pid_key='pid'):
         :type pid_key: str
         """
         df = df[[name_key, pid_key]]
-        self._tasks_by_name = df.set_index(name_key)
-        self._tasks_by_pid = df.set_index(pid_key)
+        self._tasks_by_pid = df.drop_duplicates(subset=pid_key, keep='last')\
+                .rename(columns={
+                    pid_key : 'PID',
+                    name_key : 'TaskName'})\
+                .set_index('PID').sort_index()
 
     def getTaskByName(self, name):
         """
         Get the PIDs of all tasks with the specified name.
 
+        The same PID can have different task names, mainly because once a task
+        is generated it inherits the parent name and then its name is updated
+        to represent what the task really is.
+
+        This API works under the assumption that a task name is updated at
+        most one time and it always considers the name a task had the last time
+        it has been scheduled for execution in the current trace.
+
         :param name: task name
         :type name: str
+
+        :return: a list of PID for tasks which name matches the required one,
+                 the last time they ran in the current trace
         """
-        if name not in self._tasks_by_name.index:
-            return []
-        if len(self._tasks_by_name.ix[name].values) > 1:
-            return list({task[0] for task in
-                         self._tasks_by_name.ix[name].values})
-        return [self._tasks_by_name.ix[name].values[0]]
+        return self._tasks_by_pid[self._tasks_by_pid.TaskName == name]\
+                   .index.values.tolist()
```

Comment:
When chaining methods and stuff like this, can we use implicit continuation like this

```
        return (self._tasks_by_pid[self._tasks_by_pid.TaskName == name]
                   .index.values.tolist())
```

Not sure if it's normal but I find escaped newlines really upsetting :P